### DISCUSSION TITLE
[Roadmap] Ideas for 2022 Q3 Cycle 2 Prioritization

### DISCUSSION THREAD

**Discussion by iamronen** [opencollective/opencollective#5741] [Ideas]
This is the discussion to begin surfacing ideas for Q3S2.

## How to use this discussion
Surfacing an idea looks very similar to ideas that have been shared in previous discussions we ask that you think about _problems_, (not solutions) and the impact of those problems.  Consider the following questions:

- Who is your user?
- What are they trying to achieve?

## Strategic Priorities

- #5488
- #5487
- #5486 

It's useful to use the following format when trying to reflect this:

> When ____ , I want to ___ , so I can ___

We ask submitters to select problems that have both a high degree of impact and a high degree of alignment with our focus areas. If there's an existing issue, please reference it, if not please [click here to create one](https://github.com/opencollective/opencollective/issues/new?assignees=&labels=&template=ideas--irritations-and-needs.md&title=).

We encourage everyone to surface ideas _as early as possible_ so that we can start working toward a project _pitch_ before the next prioritisation week (July 18th). 

During the sprint (in parallel with development) we work together to create a pitch that can then be reviewed in prioritisation.

If you'd like to read more about pitches, or generally find out more about our preferred approach to development check out [Shape Up by Ryan Singer](https://basecamp.com/shapeup) (it's free)

**Comment by Betree** [comment]
## Force admin users to use 2FA for an organization #5289

**Strategic priority**: security.

Our users are asking for it and it would help me sleep at night to know that admins of https://opencollective.com/opencollective/ (root admins) must have 2FA.

**Reply by iamronen** [reply]
@betree I resonate with your desire to have this in place.

## OpenCollective Root 
I can see why you would sleep over this and I believe this can be solved internally by asking everyone at OC who has root action privileges to activate 2FA. This requires no development, just a bit of organizational coordination.

## Activating 2FA 

While looking into this issue I activated 2FA on my OC account and in doing so notice that:
1. I rarely login to OpenCollective and so will rarely be prompted to authenticate at all, let alone with 2FA.
2. Since we use magic links for login, I did not have an account for OC in my password manager and had to create one in order to store the recovery keys. 
3. I, as a user, am aware of 2FA, but rarely interact with it. I suspect other users may have even less interaction with it then I did. I suspect that many non-technical users may arrive at the recovery keys screen and not know what to do with it. Practically what that means is that there, if users are forced into this without an awareness of what it is, we may end with users who have 2FA activated without having stored recovery codes. 

## Forcing Feature

1. I am not sure that adding an optional forcing feature will lead to a desired outcome of organizational accounts activating it.
2. In fact I suspect that only (a few?) security-conscious organizations will actually use a feature like this,
3. I suspect that others will be wary of forcing admins to do anything, especially if it means potentially disrupting their ability to work on the platform.

I wonder, therefor, if a more effective campaign would involve:
1. A social awareness campaign.
2. Adding in organization settings -> team a "2FA is active" indicator that enables an admin who does want to activate this in their org to see who has and who hasn't and to reach out and ask those that haven't to enable it.

**Reply by Betree** [reply]
> I can see why you would sleep over this and I believe this can be solved internally by asking everyone at OC who has root action privileges to activate 2FA. This requires no development, just a bit of organizational coordination.

This is what we do today. Unfortunately, someone in the team deactivating 2FA or a new member added by someone who didn't pay attention would go unnoticed. We ended up in this situation recently, with 3 root accounts that didn't have 2FA enforced.

> Since we use magic links for login, I did not have an account for OC in my password manager and had to create one in order to store the recovery keys.

You should never store your recovery key in your password manager as it defeats the base purpose of 2FA: if some hacks into your password manager, they'll have both your recovery keys and your password. Since we're using magic links it's ok at this point, but as soon as we implement https://github.com/opencollective/opencollective/issues/1738 it will become a real concern.

> Practically what that means is that there, if users are forced into this without an awareness of what it is, we may end with users who have 2FA activated without having stored recovery codes.

That can happen indeed, but I don't feel like our flow is very different from what's done by Google, Facebook, Github, etc. What do you suggest to improve things there? If you want to extend the thread, I suggest we create a new `[Feedback] Two factor authentication (2FA)` issue to keep the conversation focused here.

> I am not sure that adding an optional forcing feature will lead to a desired outcome of organizational accounts activating it.

As you can see in the linked issue, users are asking for it. Apart from them, I'm almost certain we'll want to enable it for **all** large hosts: foundation, europe, opensource, accountable, SCN...etc. Sounds like a big impact to me.

> A social awareness campaign.

That could be interesting indeed.

> Adding in organization settings -> team a "2FA is active" indicator that enables an admin who does want to activate this in their org to see who has and who hasn't and to reach out and ask those that haven't to enable it.

Such a page could be a nice addition, but it won't bring the same security benefits mentioned in this issue. Plus, it puts the burden on the admins rather than the system. I don't think host admins want to spend their time chasing people to ask them to enable 2FA.

Also note that "forcing 2fa" for an organization is a very common practice, among the companies that already do it: Github, Sentry, Facebook, Heroku. I'm actually pretty sure any large service that offers organization accounts has this feature in place.

**Reply by iamronen** [reply]
added to prioritization, but this needs to be better understood before we can commit to developing anything.

**Comment by shannondwray** [comment]
## #5097 More Social icons at the top of collective profiles

This would help with communication

**Reply by iamronen** [reply]
I would link this into #5680

**Reply by BenJam** [reply]
adding the profile page revamp project to the priorities for next sprint.

**Comment by kaylarep** [comment]
## https://github.com/opencollective/opencollective/issues/4564 Showcasing 'Updates' as a 'feed'

There are couple ideas presented in this issue, but ultimately any solution that consolidates Updates from multiple collectives together and makes them easier to view all together would be amazing for amplification. There's very rich content in many Updates

**Reply by Memo-Es** [reply]
I've been thinking about redesigning the way we display updates for them to be less financial-looking and more social-looking, if that makes sense. Right now, they are too close to transaction components, with a layout unsuitable for displaying media content, for example. Which, in my opinion, would be an excellent thing for engagement. Having that done, we then need to design the right way to have this on a public page, on a 'global' level (All OC Collectives), 'Host' level (Updates coming from hosted Collectives), and 'Collective' level, that would be basically the redesign of the current 'Updates' section in the Collective page.

If we do grab this one, I would like to tackle it together with #1445 :)

**Reply by iamronen** [reply]
added to board for shaping

**Comment by natehn** [comment]
## https://github.com/opencollective/opencollective/issues/5769 Collective admins can create pending contributions. 

I think this could be very impactful! Right now we are using spreadsheets and tracking tons and tons of emails to do this.

**Reply by natehn** [reply]
Part of why I was asking for https://github.com/opencollective/opencollective/issues/4927 was because of the things this could help address.

**Reply by natehn** [reply]
https://github.com/opencollective/opencollective/issues/5453 is also related

**Reply by kaylarep** [reply]
would this also be related? - for tracking another type of expense (off-platform contributions) without needing external spreadsheets - and getting a more accurate view of a group's true budget https://github.com/opencollective/opencollective/issues/1895

**Reply by alanna** [reply]
No that's a different issue @kaylarep - that's for non-money contributions (eg time volunteering, mentoring), while this is about setting up pending money contributions.

**Reply by kaylarep** [reply]
ok thanks, will add it separately

**Reply by iamronen** [reply]
@BenJam please look at this

**Reply by iamronen** [reply]
added to board - needs to be better understood - what would it take to make this whole and what implications there may be

**Reply by BenJam** [reply]
I think we already have this, no?

**Reply by alanna** [reply]
@BenJam what do you mean we already have this?

**Reply by natehn** [reply]
I commented also in the issue itself. We do not already have this - but yes, this would expand an existing feature.

**Reply by natehn** [reply]
And potentially, to some degree and depending on the design, bring some of the functionality of adding funds into the pending contribution side of things, for Host admins.

**Comment by natehn** [comment]
## https://github.com/opencollective/opencollective/issues/5458 Contribution and Expense Methods: Add Venmo option (through PayPal)

I think it is worth our time to scope out how much work this would be - for both contributions **and** expenses.

**Reply by iamronen** [reply]
added as surfaced idea that requires more attention (especially given our recent troubles with PayPal?)

**Comment by alanna** [comment]
## #4747 Extra reminders for missed pending expenses

This is a simple one (just re-send the pending expense notification to admins gain after 7 days if it hasn't been approved or rejected) that I think would help Collectives get more of their money spent, and improve the experience for people who otherwise are left hanging waiting for their expense payment. I think it fits with both communication and simplification.

**Reply by iamronen** [reply]
added to the prioritization board

**Comment by alanna** [comment]
## #2694 Ability to report undesirable comments, orders and profiles 

We have done some good work in the area of reducing spam and fraud, but we haven't fully implemented the system we envisioned where users can report all kinds of content. I think if we continue to scale, this will be more and more of an issue. I suppose this fits under "communication".

**Reply by iamronen** [reply]
I've added this to the board as "surfacing". This is a large project that requires major product attention.

**Comment by alanna** [comment]
## #5229 Allow all admins to see draft expenses

This would address simplification and help users have a smoother experience. I don't see any reason to hide draft expenses from admins.

**Reply by iamronen** [reply]
I've added this to the prioritization board.

**Comment by alanna** [comment]
## Add Financial Overview Visualizations to Collective Profile Pages #5704

This proposal is shaping up really well and could be a huge step forward for transparency and ability for Collectives to manage and communicate their budgets. When we built the host report dashboard we always imagined something similar for Collectives, and now maybe it's time.

**Reply by iamronen** [reply]
this has already been prioritized for Q3S1, though it looks like we do not have the resources to execute on it currently and so it may have to be re-proposed for the next cycle.

**Reply by BenJam** [reply]
I dare say that this could run into the next sprint so still useful to have it represented here.

**Reply by Memo-Es** [reply]
+1

**Comment by alanna** [comment]
## Fees page dropped from application flow to /foundation #5685

OCF needs the fees info to be included in the application flow. Furthermore, we've talked about using the OCF form as a template for other hosts who want custom applications, so we should ensure it's sound.

**Reply by iamronen** [reply]
added to prioritization board

**Reply by BenJam** [reply]
seems like a papercut to me (page routing incorrect) rather than a big thing. paging @znarf

**Reply by alanna** [reply]
@Betree can you look since @znarf is away?

**Reply by znarf** [reply]
It doesn't seem to be surfaced in current sprint (Q3S1) prioritized issues.

**Comment by alanna** [comment]
## Integrate with accounting software

This is requested pretty much every time a new host joins or considers joining, and we lose potential hosts on this issue alone. We have a range of options we could take in this directions, from MVP to more complex features. But if `integrations` is a priority, this surely is one of the top requested.

* #3983
* #4396
* #5670

**Reply by alanna** [reply]
The situation happening at OCF right now makes me think this is incredibly important for us to prioritise. While that problem happened outside the platform, this is the area where OC as software could help the most in preventing those kinds of problems.

**Reply by alanna** [reply]
Just had a potential third party host (large org already fiscally sponsoring multiple projects) say they would have switched to OC but no Quickbooks integration is a dealbreaker.

**Reply by natehn** [reply]
Quickbooks is the one I hear mentioned a lot too, and when I say that it doesn't have a QB integration people are visibly disappointed. Adding the integration would make a big difference for hosts and could lead to increased revenue through new hosts.

**Reply by BenJam** [reply]
bringing a conversation over from slack:

while I agree with many of the benefits that integrating with QB would offer I am not sure that the current situation at Open Collective Foundation would be solved. Open Collective does not model multiple holding accounts (bank account, PayPal, Wise, Stripe Issuing Balance etc.) as a function of the host's total holdings, so transactions between them are of no concern to Open Collective currently. To solve for this we would (I think) need to model those accounts and potentially the policy that we employ today, for example:

- there are $10k of expenses to pay, $8k are bank transactions, $2k are Paypal
- the balance in Wise is $20k and Paypal has $1k so I choose to top up Paypal with 2k as a float

Open Collective doesn't currently mandate any rules around topping up those holding accounts, so in addition to integrating QuickBooks we would need to build some capacity for Open Collective to recognise those top up transactions as moving money around inside the host in order to reconcile them against transactions that come in with from the bank account through QuickBooks. 

Either way I think this is going to need a fair amount of work to scope out so, if we think this is likely to be our biggest priority for next quarter (or more) let's get started...

**Reply by alanna** [reply]
Yes a full feature for this would be a big project. But I think it's worth digging into and scoping to see if there are any MVP ideas and to get a sense of what's actually needed from Hosts. Almost all potential new Hosts ask "How do we integrate OC with Quickbooks/Xero?" and our own hosts have to solve the same issue, so I do think it's important.

**Reply by BenJam** [reply]
> How do we integrate OC with Quickbooks/Xero

Is a solution looking for a problem. What do they actually want to achieve? That transactions from Open Collective appear in QuickBooks to reconcile against their bank account? Do they want to automatically allocate incoming transactions in Open Collective? Do they need to do the same for expenses?

I need to complete this picture in order to push the project forward. Introductions etc. appreciated but 'integrate plz' doesn't provide enough context to solve the problem.

**Reply by alanna** [reply]
@BenJam they want to fulfil their tax reporting/audit obligations as easily as possible. For 99% of organizations that amounts to using an accounting package like Quickbooks and making sure 100% of their transactions are coded in it properly. Hosts currently have to deal with OC and their accounting systems in parallel and they don't talk to each other, so keeping them synced is a lot of work, and prone to painful errors (like the recent OCF situation).

**Comment by alanna** [comment]
## Manually subscribe to Updates #1445

I think this is one of the highest-value features for the `communications` priority. Email newsletters are one of the core tools for community engagement and communication. While our Updates feature is simple, it achieves most of what many groups need, but this one thing is holding it back. Most Collectives I'm a part of have to maintain a separate newsletter list because they want to reach non-contributors too. If we offered this feature, I think groups would get even more investing in OC as their audience would be built up on our platform.

**Reply by BenJam** [reply]
+1 to this, we know that many open source supporting organisations currently support projects with time from their staff. It would also help if the company is supporting a project off the platform through something like Patreon. Would love for this to be included in https://github.com/opencollective/opencollective/discussions/5745

**Reply by iamronen** [reply]
added to board for shaping

**Comment by alanna** [comment]
## Improve Root actions

Anything that frees up dev and ops team capacity is great, and these actions would help simplify our support workflows.
* #5771
* #5651
* #5374
* #5000
* #4848
* #4841
* #4805
* #4714
* #4684
* #4648
* #4627
* #3981

If we're taking this on, we should probably also look at: Add a special permissions group for members of the support staff #4628

**Reply by BenJam** [reply]
most frequently I find myself needing to add myself as an admin to make some changes/contriobutions on a collective's/fund's behalf... But I've never seen that come up ¯\_(ツ)_/¯

**Reply by alanna** [reply]
Yes I need that too @BenJam - can you raise an issue for it?

**Reply by BenJam** [reply]
FYI I added this to the prioritisation board https://github.com/opencollective/opencollective/issues/5906

**Comment by alanna** [comment]
## Host admin can unhost Collective #3330

We ended up putting a bandaid on this by adding it to root actions, but it's still a big compliance problem waiting to happen for all other hosts.

**Comment by alanna** [comment]
## Incognito add funds #5453

This one is a process that's currently extremely complex, but could be simplified quite easily. Host admins should be able to select incognito when adding funds.

**Comment by kaylarep** [comment]
## https://github.com/opencollective/opencollective/issues/1895 $0 expenses / non-financial contributions
for non-money contributions (eg time volunteering, mentoring)

Ultimately, this is about having a Collectives' page be an accurate reading of their real budget - it should include making space for these extraneous off-platform expenses 
e.g.:
- non-financial contributions, 
- recording retroactive contributions from before they signed up on OC, 
- being able to record off-platform contributions into the platform

for tracking other type of expense without needing external spreadsheets - and getting a more accurate view of a group's true budget

**Reply by BenJam** [reply]
would be great to do this, we did the same from 24pullrequests.com a few years back after we realised that contributing code wasn't the best way to support a project sometimes.

**Reply by iamronen** [reply]
added this to the prioritization board, though this is just surfaced and needs more shaping before we can commit to development.

**Comment by alanna** [comment]
## Move hosted accounts to a separate section #5457

Treating hosted Collectives as "contributions" is confusing and makes the design of host profile pages odd to manage. It would simplify things to give this function its own section. 

There was a suggestion in there to rethink the whole design of the host pages... if that's going to be prioritized now, then ok, but I think this smaller change is still worth doing now even if a whole redesign is not happening in the short term.

**Reply by iamronen** [reply]
yes, this belongs in #5680

**Comment by kaylarep** [comment]
## Allow virtual cards receipts to be uploaded while processing https://github.com/opencollective/opencollective/issues/5783 

Suggestion from user-feedback.

They are finding it difficult to keep up-to-date with uploading receipts. Their current processes involves them remembering to check back in 2-4 days after transaction.

**Reply by alanna** [reply]
@kewitz is there anything technical preventing this now?

**Reply by alanna** [reply]
I think that has already been done?

**Reply by BenJam** [reply]
just checked it, nope.  It's on the parent project for Virtual Cards.

**Reply by BenJam** [reply]
FYI I added the entirety of the VC feedback issue to  prioritisation

**Comment by alanna** [comment]
## Attachment previews not working #4993

This being broken prevents the expense payment workflow from being simple and smooth, because you have to download the attached files to see receipts.

**Reply by iamronen** [reply]
added to board as pitching.

**Comment by alanna** [comment]
## Do not treat system-generated expense summary as an attachment #3865

Another one for simplification. It's just confusing and slows down the process of reviewing actual expense attachments.

**Comment by kaylarep** [comment]
## Virtual Card Edits https://github.com/opencollective/opencollective/issues/5410

by now, we are getting more clarity about how we/users use virtual cards. Would be nice to tackle some of the issues/feedback from this issue.

Apologies for this issue being a mess, if we choose to prioritize this, I will plan to reorganize it all as separate issues within a Project

**Reply by iamronen** [reply]
@BenJam please have a look at this,
There are plenty of well-understood needs here but this needs some product attention in order to shape it into a coherent, well bounded, perhaps internally prioritized project. I have a feeling that the things in here can quickly add up to a rather large project.
I've added it to the board as requires shaping.

**Reply by BenJam** [reply]
I closed this one in favour of the feedback issue which i linked all the feedback up into, worth having a session on this before going any further.

**Comment by alanna** [comment]
## Attach files/notes to Collective #4894

This would really help with simplification and communication for us as a host admin team. We have to track lots of things like contracts and grants associated with a Collective, and linking docs directly would be super helpful.

**Reply by iamronen** [reply]
added to prioritization - requires shaping

**Reply by alanna** [reply]
For MVP even a simple text box only visible to Host admins would be quite useful, where we can put notes and links to files in Drive etc.

**Comment by SudharakaP** [comment]
## Add Further Filtering for the `/search` page https://github.com/opencollective/opencollective/issues/5793

This has been suggested several times although I don't think we have a clear consensus on what we want to achieve here. Nevertheless maybe a discussion to scope this out might be helpful. 🤔

**Comment by kaylarep** [comment]
## Improve Search https://github.com/opencollective/opencollective/issues/5809

There was a previously-suggested proposal to do this, but we decided to first do a basic improvement of our search feature within the existing framework. 

This issue has come up a couple times since that MVP, so it's perhaps time to move this one along.

**Comment by kaylarep** [comment]
## https://github.com/opencollective/opencollective/issues/5689 Collective cards in the 'Contributions' section appear as they do in search queries - with Tag descriptors

Currently the collective cards in this section do not share much information. This is one of the only places on the platform where all of a host's Collective's can be viewed all in one place 'at-a-glance'

Will also help amplify Collectives (Communication strategy)

**Comment by alanna** [comment]
## Host admin can set date when confirming contribution #5693

This is a small thing that would be very helpful to simplify accounting processes.

**Comment by alanna** [comment]
## Record merchant account timestamps #5817

This is a request coming from OCF's accountants in relation to improving internal control processes.

**Comment by Memo-Es** [comment]
## Onboarding flow - Find a host https://github.com/opencollective/opencollective/issues/5672 

Follow-up with the work to improve our onboarding flow, specifically with the experience to find a host, embedded inside the onboarding flow for Collective admins.

**Reply by BenJam** [reply]
added to the board under the onboarding project.

**Comment by Memo-Es** [comment]
## Tiers follow-up work
 - Custom image for tiers
 - Tiers for sponsor organizations setting

*This one is part of #5732

**Comment by alanna** [comment]
## Bug fixing 🐛

We have around 130 [open bug issues](https://github.com/opencollective/opencollective/issues?q=is%3Aissue+is%3Aopen+label%3Abug). Let's consider if we want to set aside space to squash some. It's hard to know how to group them but here are some I think are clearly scoped and either quick easy wins or especially causing pain (and not already being worked on actively). If others want to suggest their own pet bugs 🪲 please do.

- #5820 
- #5751 
- #5749 
- #5728 
- #5719
- #5685 
- #5659 
- #5490
- #5473
- #5452
- #5438 
- #5392
- #5359
- #5286
- #5228 
- #5207 
- #5180 
- #5101
- #5077
- #5068
- #4988
- #4953

**Comment by alanna** [comment]
## Improve validations in the expense form

Bringing this one @Betree previously suggested [in his words] as it's still causing a cluster of related issues that create a very un-simple user experience flow for expenses.

Complexity: 1-3 days (medium)

We had many issues in the past with the expense flow being unexpectedly blocked and it's still happening to us in different situations. This started after the introduction of grants, where we started disabling the Cancel/Prev/Next buttons if there was an error in the form (even if for some reason the error was made on another step).

The following issues result from this root cause:
- https://github.com/opencollective/opencollective/issues/5020
- https://github.com/opencollective/opencollective/issues/5222
- https://github.com/opencollective/opencollective/issues/5429
- Edit: https://github.com/opencollective/opencollective/issues/5306 also originates from this

It's a really annoying bug because you're basically trapped when it happens to you, everything is disabled and there's no message to explain what's wrong. I had started https://github.com/opencollective/opencollective-frontend/pull/7533 to solve this and would like to take some time to finish it.

**Reply by BenJam** [reply]
this is already on the board for this sprint https://github.com/orgs/opencollective/projects/5/views/24

**Comment by Memo-Es** [comment]
## #5680 Collective page revamp

This is a significant issue because it adds value to two of our most essential flows: Contribution flow #5732 and onboarding flow #4829. We have been discussing differentiating our profiles, which is the first step.
I estimate two sprints back to back to work on this one, the first one right after the retreat focused on the final definition and design, and the second one to implement it.

**Comment by alanna** [comment]
## Projects and Events can have their own different admins #4611

I think this relatively small change would unlock a lot of potential and users keep requesting it. It speaks to the communication and simplification priorities because it would enable people running specific events and projects to get notifications and take actions relevant to their own areas. Right now unless a Collective wants to make someone an admin of their whole thing, the current admins have to take a lot of actions on behalf of the actual organisers of the event/project and it's messy.

**Comment by kaylarep** [comment]
## Custom payees without invitations for invoices https://github.com/opencollective/opencollective/issues/5894

Submitted from user feedback

>we need to be able to add expenses easily to our collective's page. We can't invite companies to be "payees" when adding an invoice. . For example, we use DigitalOcean for hosting/infrastructure. They charge us automatically every month. We can't invite them, but we need to track that expense.

**Comment by iamronen** [comment]
This thread is CLOSED for proposals. 

![image](https://user-images.githubusercontent.com/158833/186710929-5a7ff131-d638-47e5-8d90-2a4b61e32b7c.png)

Additional proposals can be submitted in the [discussion thread for Q4S1](https://github.com/opencollective/opencollective/discussions/5899)

**Reply by kaylarep** [reply]
thanks for adding this note!
